controllers/topics/index.js

```diff
@@ -184,14 +189,38 @@ const followTopicV2 = async (req, res) => {
     let data = [];
     if (getTokenUserStatus) {
       await userTopicService.unfollowTopic(detailTokenUser.user_id, topic_id);
-      data.push(await _afterPutTopic(true, token, detailTokenUser.user_id, name));
+      data.push(
+        await _afterPutTopic(
+          true,
+          token,
+          detailTokenUser.user_id,
+          name,
+          detailTokenUser.is_anonymous
+        )
+      );
     } else {
       await userTopicService.followTopic(detailTokenUser.user_id, topic_id);
-      data.push(await _afterPutTopic(false, token, detailTokenUser.user_id, name));
+      data.push(
+        await _afterPutTopic(
+          false,
+          token,
+          detailTokenUser.user_id,
+          name,
+          detailTokenUser.is_anonymous
+        )
+      );
 
       if (getSecondUserStatus) {
         await userTopicService.unfollowTopic(secondDetailUser.user_id, topic_id);
-        data.push(await _afterPutTopic(true, token, secondDetailUser.user_id, name));
+        data.push(
+          await _afterPutTopic(
+            true,
+            token,
+            secondDetailUser.user_id,
+            name,
+            detailTokenUser.is_anonymous
+          )
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here seems to be correct, but it's a bit hard to read due to the repetition of similar blocks of code. Consider refactoring this section to reduce redundancy and improve readability. Here is a suggestion:

```diff
-      if (getTokenUserStatus) {
-        await userTopicService.unfollowTopic(detailTokenUser.user_id, topic_id);
-        data.push(
-          await _afterPutTopic(
-            true,
-            token,
-            detailTokenUser.user_id,
-            name,
-            detailTokenUser.is_anonymous
-          )
-        );
-      } else {
-        await userTopicService.followTopic(detailTokenUser.user_id, topic_id);
-        data.push(
-          await _afterPutTopic(
-            false,
-            token,
-            detailTokenUser.user_id,
-            name,
-            detailTokenUser.is_anonymous
-          )
-        );
-
-        if (getSecondUserStatus) {
-          await userTopicService.unfollowTopic(secondDetailUser.user_id, topic_id);
-          data.push(
-            await _afterPutTopic(
-              true,
-              token,
-              secondDetailUser.user_id,
-              name,
-              detailTokenUser.is_anonymous
-            )
-          );
-        }
-      }

+      const users = [
+        { status: getTokenUserStatus, details: detailTokenUser },
+        { status: getSecondUserStatus, details: secondDetailUser },
+      ];
+
+      for (const user of users) {
+        if (user.status) {
+          await userTopicService.unfollowTopic(user.details.user_id, topic_id);
+        } else {
+          await userTopicService.followTopic(user.details.user_id, topic_id);
+        }
+
+        data.push(
+          await _afterPutTopic(
+            user.status,
+            token,
+            user.details.user_id,
+            name,
+            user.details.is_anonymous
+          )
+        );
+      }
```
This refactoring reduces the code duplication and makes it easier to add more users in the future if needed.



<!-- This is an auto-generated comment by OSS CodeRabbit -->